pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -62,7 +62,8 @@ func (pl *TaintToleration) EventsToRegister() []framework.ClusterEvent {
 
 // Filter invoked at the filter extension point.
 func (pl *TaintToleration) Filter(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
-	if nodeInfo == nil || nodeInfo.Node() == nil {
+	node := nodeInfo.Node()
```

Comment:
I don't understand why the null pointer check is useless, get a value from the nil pointer will definitely get the `invalid memory address or nil pointer dereference` exception.

Even this value pass here won't be be nil atm, the nil pointer check is still better and we cannot assume it will never be nil.

Reply:
let's see the method here https://github.com/kubernetes/kubernetes/blob/d14ba948ef63769e9767aebd5a08171832a1bbf6/pkg/scheduler/framework/types.go#L534-L539

Reply:
okay, my bad, missed the pr description.
anyway, this could be more concise
```suggestion
	if nodeInfo.Node() == nil {
```
but I am fine with this too.

Reply:
I think he wants to reuse this.